scripts/space-provider-config

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+# This is a helper script to set the space provider access from config file
+# as Config value in space provider yaml.
+
+# Check if two parameters are provided
+if [ $# -ne 2 ]; then
+    echo "Usage: $0 <config file> <provider yaml>"
+    exit 1
+fi
+
+config="$1"
+provider="$2"
+
+# Check if config file exists
+if [ ! -f "$config" ]; then
+    echo "Error: $config does not exist."
+    exit 1
+fi
+
+# Check if provider file exists
+if [ ! -f "$provider" ]; then
+    echo "Error: $provider does not exist."
+    exit 1
+fi
+
+# Define the specConfig string
+specConfig="  Config: |"
+
+# Check if config field exists in the provider yaml file and remove text after it
+if grep -q "$specConfig" "$provider"; then
+    sed -i "/$specConfig/,\$d" "$provider"
```

Comment:
You should check out `sed -i` on both Mac and Linux.

Reply:
In particular, the `-i` flag with no actual suffix is handled differently in the two.

Reply:
Mustfix: this will not work on a Mac, where `-i` MUST be followed by an extension.

```console
mspreitz@mjs12 ~ % man sed | head -10
SED(1)                       General Commands Manual                      SED(1)

NAME
     sed – stream editor

SYNOPSIS
     sed [-Ealnru] command [-I extension] [-i extension] [file ...]
     sed [-Ealnru] [-e command] [-f command_file] [-I extension] [-i extension]
         [file ...]
```

Reply:
I believe changing the line to sed -i "" "/$specConfig/,\$d" "$provider" should work on both gnu sed and bsd sed. but I need to find a machine to test this on.

Reply:
Replaced `sed` with `awk`. Tested on both Mac and Linux.